rfcs/20240311-gather-scatter-batching-dims.md

```diff
@@ -0,0 +1,456 @@
+    of `start_indices` corresponding to `index_vector_dim` is not included.
+  * `offset_dim_sizes = slice_sizes` except that the dimension sizes in
+    `slice_sizes` corresponding to `collapsed_slice_dims` and
+    `operand_batching_dims` are not included.
+  * `combine` puts `batch_dim_sizes` at axes corresponding to `batch_dims` and
+   `offset_dim_sizes` at axes corresponding to `offset_dims`.
+* (C23) `element_type(operand) = element_type(result)`.
+
+#### Examples
+
+```mlir
+// %operand: [
+//            [
+//             [[1, 2], [3, 4], [5, 6], [7, 8]],
+//             [[9, 10],[11, 12], [13, 14], [15, 16]],
+//             [[17, 18], [19, 20], [21, 22], [23, 24]]
+//            ],
+//            [
+//             [[25, 26], [27, 28], [29, 30], [31, 32]],
+//             [[33, 34], [35, 36], [37, 38], [39, 40]],
+//             [[41, 42], [43, 44], [45, 46], [47, 49]]
+//            ]
+//           ]
+// %start_indices: [
+//                  [
+//                   [[0, 0], [1, 0], [2, 1]],
+//                   [[0, 1], [1, 1], [0, 9]]
+//                  ],
+//                  [
+//                   [[0, 0], [2, 1], [2, 2]],
+//                   [[1, 2], [0, 1], [1, 0]]
+//                  ]
+//                 ]
+%result = "stablehlo.gather"(%operand, %start_indices) {
+  dimension_numbers = #stablehlo.gather<
+    offset_dims = [3, 4],
+    collapsed_slice_dims = [1],
+    operand_batching_dims = [0],
+    start_indices_batching_dims = [1],
+    start_index_map = [2, 1],
+    index_vector_dim = 3>,
+  slice_sizes = array<i64: 1, 1, 2, 2>,
+  indices_are_sorted = false
+} : (tensor<2x3x4x2xi32>, tensor<2x2x3x2xi64>) -> tensor<2x2x3x2x2xi32>
+// %result: [
+//           [
+//            [
+//             [[1, 2], [3, 4]],
+//             [[3, 4], [5, 6]],
+//             [[13, 14], [15, 16]]
+//            ],
+//            [
+//             [[33, 34], [35, 36]],
+//             [[35, 36], [37, 38]],
+//             [[41, 42], [43, 44]]
+//            ]
+//           ],
+//           [
+//            [
+//             [[1, 2], [3, 4]],
+//             [[13, 14], [15, 16]],
+//             [[21, 22], [23, 24]]
+//            ],
+//            [
+//             [[43, 44], [45, 46]],
+//             [[33, 34], [35, 36]],
+//             [[27, 28], [29, 30]]
+//            ]
+//           ]
+//          ]
+```
+
+&nbsp;[More Examples](../stablehlo/tests/interpret/gather.mlir)
+
+### scatter
+
+#### Semantics
+
+Produces `results` tensors which are equal to `inputs` tensors except that
+several slices specified by `scatter_indices` are updated with the values
+`updates` using `update_computation`.
+
+The following diagram shows how elements in `updates...` map on elements in
+`results...` using a concrete example. The diagram picks a few example
+`updates...` indices and explains in detail which `results...` indices they
+correspond to.
+
+![](images/20240311-gather-scatter-batching-dims/scatter.svg)
```

Comment:
In the image `start_indices` -> `scatter_indices`
`operand_batching_dims` -> `input_batching_dims`

Reply:
Done. Thanks for pointing out!